Sources/DVR/SessionDataTask.swift

```diff
@@ -97,7 +101,7 @@ final class SessionDataTask: URLSessionDataTask {
             }
             
             // Create interaction
-            this.interaction = Interaction(request: this.request, response: response, responseData: data)
+            this.interaction = Interaction(request: this.request, response: response, responseData: data, filter: this.session.filter)
```

Comment:
My gut tells me we should not associate a filter with an interaction like this.

Reply:
I can explain my thought process better during our call.